cirq/work/observable_measurement.py

```diff
@@ -0,0 +1,210 @@
+# Copyright 2020 The Cirq developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import dataclasses
+import itertools
+from typing import Iterable, List, Tuple, TYPE_CHECKING, Sequence, Dict
+
+import numpy as np
+import sympy
+
+from cirq import circuits, ops, value
+from cirq.work.observable_settings import (
+    InitObsSetting,
+    _MeasurementSpec,
+)
+
+if TYPE_CHECKING:
+    import cirq
+    from cirq.value.product_state import _NamedOneQubitState
+
+
+def _with_parameterized_layers(
+    circuit: 'cirq.Circuit',
+    qubits: Sequence['cirq.Qid'],
+    no_initialization: bool,
```

Comment:
The lack of symmetry between initial and final rang some warning bells for me here, though I understand that the final layer will be always required (even in case of ZZZZZZ...) to implement the flips. 
However, we could just have the inital layer always defaulting to KET_ZERO...and as long as the compilers compile away negligible gates, then we don't have to have this extra arg that's carried across all the function calls.

Reply:
The theory here is that the user will have already compiled their circuit and we only add single-qubit gates that we presume to be in the hardware-native gateset, so there's no room for a negligibility compiler. Or at least we wouldn't want to do that in case it disturbs how the user supplied `circuit`. 

It's true that with these utility functions this boolean has to be passed around a lot. I think this is just the reality of the situation where you sometimes have to have rotations at the beginning. We don't want to add them indiscriminately, as it could have performance implications. Of course, these are all private functions and the appropriate value for `no_initialization` will be deduced from user-supplied input in the next PR and handled transparently :)